ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Genesis/Governor.hs

```diff
@@ -49,16 +70,16 @@ sharedCandidatePrefix ::
   SecurityParam ->
   AnchoredFragment (Header blk) ->
   Map peer (AnchoredFragment (Header blk)) ->
-  AnchoredFragment (Header blk)
+  (AnchoredFragment (Header blk), Map peer (AnchoredFragment (Header blk)))
 sharedCandidatePrefix (SecurityParam k) curChain candidates =
-  trunc
+  (trunc, suffixes)
   where
     trunc | excess > 0 = snd (AF.splitAt excess shared)
           | otherwise = shared
 
     excess = AF.length shared - fromIntegral k
```

Comment:
Splitting the result with `excess` here is causing the loe fragment to not intersect with the current selection.

What is the purpose of trimming the loe fragment like this?

Reply:
I removed the `excess` computation in e455562. I believe the old documentation of the function was trying to justify it, but it was unintelligible to me, and the old behavior wasn't working well with chain selection.